tech-specs/src/explore/design/dkg.md

```diff
@@ -1 +1,154 @@
+off-chain communication. However, the entire DKG cannot exist solely in the 
+gossip layer as consensus on the final shared secrets must be achieved. This 
+will require posting information on chain.
+
+## Starting the Protocol
+
+To initializing the protocol to generate a new key, the  current block 
+proposer proposes a new instance. This would be posted on the blockchain and
+notify all the participating validators to begin the announcement phase. 
+However, when should the proposer make this decision? In all likelihood it will be a combination of blocks transpired since
+the last instance completed as well as special cases (such as an instance
+failing to complete).
+
+## Announcing and distributing weight shares
+
+In order for the DKG protocol to work, it requires the contribution of 
+ \\(\frac{2}{3} \\) of the total staked weight among live validators. In 
+order to know when a subset of validators own at least \\(\frac{2}{3}\\)
+of the total weight, it is necessary for validators to announce their stake.
+
+Furthermore, the total weight will be scaled down to a fixed parameter 
+\\( W\\) (and fractional weights rounded appropriately to acheive integral
+values). After this procedure, each validator has a corresponding number of
+weight shares. 
+
+Lastly, the DKG protocol requires a canonical ordering of validators with
+shares must be agreed upon. This will be done via the consensus layer.
+The current block proposer will apportion out weight shares as well as the
+aforementioned ordering (which induces a natural partition of the weight 
+shares) to each validator that made an announcement. This is included in the
+block. Once this block is on the blockchain, the next step of the DKG can proceed.
+
+## Dealers
+In a single DKG instance, a subset of validators holding at least
+\\( \frac{2}{3} \\) of the weight shares must each act as dealers. It
+is the responsibility of a dealer to post a signed message to the blockchain
+which includes the session id (\\( \tau \\)) as well the data of the
+PVSS (we do not concern ourselves with the details here). The creation of this
+message is handled by the Ferveo library.
+
+This means that the signed message is a transaction that must be gossiped and
+eventually included on the blockchain. For this particular transaction, there
+is a special verification that must be done, otherwise the transaction should
+not be included in any proposed block. This verification is provided by the
+Ferveo library.
+
+## Aggregation
+
+At some point, enough validators (meaning their combined weight  shares
+exceeds \\( \frac{2}{3}\\) of the total) have their PVSS on the chain 
+for the session with id \\( \tau \\). At this point, the next block proposer can
+aggregate the messages into a single instance using Ferveo. This is then 
+included in the next block. Once this DKG is on the blockchain, it can be 
+used for encryption of transactions until the next DKG instance is run.
```

Comment:
Yes, it's better to have PVSS transcripts posted to the chain, because the cost (extra on-chain storage) is outweighed by the benefits (simplicity, lower overall compute time) of synchronization - and the PVSS transcripts should be pruned from the blockchain archive once the relevant epoch is over.

Right now I don't think any off-chain gossiping is needed. Other than the PVSS transactions being gossiped to the block proposers like other txs.